queries/ja.py

```diff
@@ -0,0 +1,297 @@
+
+
+def QJaPhoneNum4NameQuery(node, params, result):
+    result.qtype = "PhoneNum4Name"
+
+
+_JA_SOURCE = "ja.is"
+
+_JA_API_URL = "https://api.ja.is/search/v6/?{0}"
+
+
+def query_ja_api(q: str) -> Optional[Dict]:
+    """ Send query to ja.is API """
+    key = read_api_key("JaServerKey")
+    if not key:
+        # No key, can't query the API
+        logging.warning("No API key for ja.is")
+        return None
+
+    qdict = {"q": q}
+    headers = {"Authorization": key}
+
+    # Send API request, get back parsed JSON
+    url = _JA_API_URL.format(urlencode(qdict))
+    res = query_json_api(url, headers=headers)
+
+    return res
+
+
+_MOBILE_FIRST_NUM = "678"
+
+
+def _best_number(item: Dict) -> Optional[str]:
+    """ Return best phone number, given a result item from ja.is API """
+    phone_num = item.get("phone")
+    add_nums = item.get("additional_phones")
+    if not phone_num and not add_nums:
+        return None
+
+    # First, see if the canonical phone number is a mobile phone number
+    # Those should be preferred
+    if phone_num and "number" in phone_num:
+        if phone_num.get("mobile") == True:
+            return phone_num["number"]
+
+    # Otherwise, try the first mobile number we find in add. phone numbers
+    if add_nums:
+        for pn in add_nums:
+            if pn and "number" in pn and pn.get("mobile") == True:
+                return pn["number"]
+
+    # OK, didn't find any mobile numbers. Just return canoncial number
+    return phone_num.get("number") if phone_num else None
+
+
+def _answer_phonenum4name_query(q: Query, result):
+    """ Answer query of the form "hvað er síminn hjá [íslenskt mannsnafn]?" """
+    res = query_ja_api(result.qkey)
+
+    nþgf = NounPhrase(result.qkey).dative or result.qkey
+
+    # Verify that we have a sane response with at least 1 result
+    if not res or not res.get("people") or not res["people"].get("items"):
+        return gen_answer("Ekki tókst að fletta upp {0}.".format(nþgf))
+
+    # Check if we have a single canonical match from API
+    single = len(res["people"]["items"]) == 1
+    allp = res["people"]["items"]
+    first = allp[0]
+    fname = first["name"]
+    if not single:
+        # Many found with that name, generate smart message asking for disambiguation
+        one_name_only = len(result.qkey.split()) == 1
+        msg = "Það fundust margir með það nafn. Prufaðu að tilgreina {0}heimilisfang".format(
+            "fullt nafn og " if one_name_only else ""
+        )
+        # Try to generate example, e.g. "Jón Jónssón á Smáragötu"
+        for i in allp:
+            try:
+                street_nf = i["address_nominative"].split()[0]
+                street_þgf = i["address"].split()[0]
+                msg = msg + " t.d. {0} {1} {2}".format(
+                    fname, iceprep_for_street(street_nf), street_þgf
+                )
+                break
+            except Exception as e:
+                print("Exception " + str(e))
```

Comment:
Better exception handling here: remove print() and add logging if unexpected exception